### ISSUE TITLE
[no-undef] False positive for Interfaces

### ISSUE THREAD

**Issue by dawsonc623** [typescript-eslint/typescript-eslint#342]
**Repro**

Paste the below code into a file and lint it.

```TS
interface StringCollection
{
  get(
    index : number
  ) : string;
}

export default StringCollection;
```

**Expected Result**

ESLint should not produce any errors

**Actual Result**

ESLint produces a `no-undef` error.

**Additional Info**

Oddly, it also complains about me using `new Map<T, U>()` with the same rule. I am not sure if it is the same root cause, but I figured it was worth noting.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.1` |
| `node`                             | `8.15.0` |
| `npm`*                              | `6.4.1` |
| `yarn`*                            | `1.12.3` |

\* I use Yarn instead of NPM

**Comment by edsrzf** [comment]
Slightly more minimal repro case:

```
interface I {}
export default I
```

Non-default exports also trigger it:

```
interface I {}
export { I }
```

Type aliases are affected too:

```
type T = {}
export default T
```

I'm going to try to figure this out because it's affecting me, but don't really know what I'm doing so any help's appreciated.

**Comment by bradzacher** [comment]
We've got plans to fix the scope analysis, but it's a big job because of the complexities that the type system introduces, so there's no ETA on that.

A workaround is to not use the `no-undef` rule with typescript code - the tsc will catch these sorts of bugs for you, so you don't really need the rule on.

**Comment by abdonrd** [comment]
Any news here? Thanks in advance!

**Comment by linonetwo** [comment]
Just disable it https://github.com/eslint/typescript-eslint-parser/issues/437

**Comment by bradzacher** [comment]
Merging into #1856
